src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1224,83 +1224,144 @@ CATCH_RETURN();
 
-    //uint32_t axisTagPresence = AxisTagPresenceNone;
-    //for (auto& fontAxisValue : axesVector)
-    //{
-    //    switch (fontAxisValue.axisTag)
-    //    {
-    //    case DWRITE_FONT_AXIS_TAG_WEIGHT:
-    //        axisTagPresence |= AxisTagPresenceWeight;
-    //        break;
-    //    case DWRITE_FONT_AXIS_TAG_WIDTH:
-    //        axisTagPresence |= AxisTagPresenceWidth;
-    //        break;
-    //    case DWRITE_FONT_AXIS_TAG_ITALIC:
-    //        axisTagPresence |= AxisTagPresenceItalic;
-    //        break;
-    //    case DWRITE_FONT_AXIS_TAG_SLANT:
-    //        axisTagPresence |= AxisTagPresenceSlant;
-    //        break;
-    //    case DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE:
-    //        axisTagPresence |= AxisTagPresenceOpticalSize;
-    //        break;
-    //    }
-    //}
+    uint32_t axisTagPresence = AxisTagPresenceNone;
+    for (auto& fontAxisValue : axesVector)
+    {
+        switch (fontAxisValue.axisTag)
+        {
+        case DWRITE_FONT_AXIS_TAG_WEIGHT:
+            axisTagPresence |= AxisTagPresenceWeight;
+            break;
+        case DWRITE_FONT_AXIS_TAG_WIDTH:
+            axisTagPresence |= AxisTagPresenceWidth;
+            break;
+        case DWRITE_FONT_AXIS_TAG_ITALIC:
+            axisTagPresence |= AxisTagPresenceItalic;
+            break;
+        case DWRITE_FONT_AXIS_TAG_SLANT:
+            axisTagPresence |= AxisTagPresenceSlant;
+            break;
+        case DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE:
+            axisTagPresence |= AxisTagPresenceOpticalSize;
+            break;
+        }
+    }
 
-    //if ((axisTagPresence & AxisTagPresenceWeight) == 0)
-    //{
-    //    axesVector.push_back({ DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight) });
-    //}
-    //if ((axisTagPresence & AxisTagPresenceWidth) == 0)
-    //{
-    //    auto ayy1 = 1;
-    //    //auto value = FontStretchToWidthAxisValue(fontStretch);
-    //    axesVector.push_back({ DWRITE_FONT_AXIS_TAG_WIDTH, gsl::narrow<float>(ayy1) });
-    //}
-    //if ((axisTagPresence & AxisTagPresenceItalic) == 0)
-    //{
-    //    auto ayy2 = 1;
-    //    //auto value = FontStyleToItalicFixedAxisValue(fontStyle);
-    //    axesVector.push_back({ DWRITE_FONT_AXIS_TAG_ITALIC, gsl::narrow<float>(ayy2) });
-    //}
-    //if ((axisTagPresence & AxisTagPresenceSlant) == 0)
-    //{
-    //    auto ayy3 = 1;
-    //    //auto value = FontStyleToSlantFixedAxisValue(fontStyle);
-    //    axesVector.push_back({ DWRITE_FONT_AXIS_TAG_SLANT, gsl::narrow<float>(ayy3) });
-    //}
-    //if ((axisTagPresence & AxisTagPresenceOpticalSize) == 0)
-    //{
-    //    auto pointSize = 1;
-    //    //const auto pointSize = DIPsToPoints(fontSize);
-    //    axesVector.push_back({ DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE, gsl::narrow<float>(pointSize) });
-    //}
+    if ((axisTagPresence & AxisTagPresenceWeight) == 0)
+    {
+        axesVector.push_back({ DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight) });

```

Comment:
Consider `emplace_back`

Reply:
I tried `axesVector.emplace_back(DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight));` and got this error:

`error C2661: 'DWRITE_FONT_AXIS_VALUE::DWRITE_FONT_AXIS_VALUE': no overloaded function takes 2 arguments`

Any idea how to fix this? This is weird because the `push_back` version works so the default constructor clearly does take in 2 arguments

Reply:
`emplace_back` forwards the given arguments 1:1 to the constructor of the vector element.
`DWRITE_FONT_AXIS_VALUE` is a C struct without an explicit constructor. You have to write
```cpp
axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight)});
```
which quite likely also works as
```cpp
axesVector.emplace_back({DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight)});
```

Reply:
This doesn't work:
`axesVector.emplace_back({DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight)});`

This one does though! Thanks
`axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight)});`

Reply:
Whoops, turns out 

`axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{DWRITE_FONT_AXIS_TAG_WEIGHT, float(fontWeight)});`

causes the error: `Don't use C-style casts (type.4)`.

Reply:
do a `gsl::narrow_cast<float>(fontWeight)` instead